src/dfx/src/commands/extension/run.rs

```diff
@@ -26,8 +25,6 @@ impl From<Vec<OsString>> for RunOpts {
 
 pub fn exec(env: &dyn Environment, opts: RunOpts) -> DfxResult<()> {
     let mgr = env.new_extension_manager()?;
-    let dfx_version = &env.get_version().to_string();
-    let path_to_dfx_cache = get_bin_cache(dfx_version)?;
-    mgr.run_extension(path_to_dfx_cache, opts.name, opts.params)?;
+    mgr.run_extension(opts.name, opts.params)?;
```

Comment:
unnecessary to pass the path to dfx cache, cause it can be derived (ExtensionManager already has a field `dfx_version`)